File: app/src/main/java/com/software/engineering/alcohollife/ui/review/WriteReview.java
package com.software.engineering.alcohollife.ui.review;

import androidx.annotation.NonNull;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.app.Notification;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.software.engineering.alcohollife.R;

import org.w3c.dom.Text;

public class WriteReview extends AppCompatActivity {
    String s;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_write_review);

        Toolbar toolbar = (Toolbar)findViewById(R.id.my_toolbar);
        setSupportActionBar(toolbar);
        ActionBar ab = getSupportActionBar();
        ab.setDisplayShowTitleEnabled(false);
        ab.setDisplayHomeAsUpEnabled(true);

        final EditText editText = findViewById(R.id.editTextTextPersonName);
        Button button = findViewById(R.id.button2);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                s = editText.getText().toString();
                review_submit(s);

            }
        });

    }

    public void review_submit(String s){
        Toast.makeText(getApplicationContext(), s, Toast.LENGTH_SHORT).show();

    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()){
            case android.R.id.home:
                finish();
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/software/engineering/alcohollife/ui/review/WriteReview.kt
package com.software.engineering.alcohollife.ui.review

import android.app.Application
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.EditText
import android.widget.RatingBar
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.lifecycle.Observer
import com.bumptech.glide.Glide
import com.software.engineering.alcohollife.R
import com.software.engineering.alcohollife.model.data.ReviewData
import com.software.engineering.alcohollife.model.data.ReviewPostData
import com.software.engineering.alcohollife.model.network.base.ApiStatus
import com.software.engineering.alcohollife.model.network.base.RestClient
import kotlinx.android.synthetic.main.activity_item_page.*
import kotlinx.android.synthetic.main.activity_write_review.*

class WriteReview : AppCompatActivity() {
    private val model = RestClient.getDrinkService()

    private val alcoholName by lazy { intent!!.getStringExtra(EXTRA_ALCOHOL_NAME)!! }
    private val imgUrl by lazy { intent!!.getStringExtra(EXTRA_IMG_URL)!! }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_write_review)

        val toolbar = findViewById<View>(R.id.my_toolbar) as Toolbar
        setSupportActionBar(toolbar)

        supportActionBar?.setDisplayShowTitleEnabled(false)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val ratingBar = findViewById<RatingBar>(R.id.rating_bar)
        val editText = findViewById<EditText>(R.id.editTextTextPersonName)

        textView.text = alcoholName

        Glide.with(this)
            .load(imgUrl)
            .into(imageView)



        button2.setOnClickListener {
            review_submit(ratingBar.rating, alcoholName, editText.text.toString())
        }
    }

    fun review_submit(rating:Float, name:String, review_text:String) {
        model.reviewPost(ReviewPostData(rating, name, review_text)).observe(this, Observer {
            when (it) {
                is ApiStatus.Success -> {
                    Toast.makeText(applicationContext, "리뷰등록이 완료되었습니다.", Toast.LENGTH_SHORT).show();
                    finish();
                }
            }
        })
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.home -> finish()
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        private const val EXTRA_ALCOHOL_NAME = "EXTRA_ALCOHOL_NAME"
        private const val EXTRA_IMG_URL = "EXTRA_IMG_URL"
        fun getStartIntent(context: Context?, alcohol: String, img: String): Intent {
            return Intent(context, WriteReview::class.java).apply {
                putExtra(EXTRA_ALCOHOL_NAME, alcohol)
                putExtra(EXTRA_IMG_URL, img)
            }

        }
    }
}